baseplate/lock.py

```diff
@@ -0,0 +1,162 @@
+"""Simple locks in memcached or redis."""
+from __future__ import absolute_import
+
+import os
+import random
+import socket
+import time
+
+from pymemcache.client.base import PooledClient
+from redis import StrictRedis
+
+
+class TimeoutExpired(Exception):
+    pass
+
+
+MIN_RETRY_WAIT = 0.1
+MAX_RETRY_WAIT = 1.0
+
+
+class WrappedMemcache(object):
+    def __init__(self, cache):
+        self.cache = cache
+
+    def add_if_unset(self, key, value, ttl):
+        did_add = self.cache.add(key, value, expire=ttl, noreply=False)
+        return did_add
+
+    def get(self, key):
+        return self.cache.get(key)
+
+    def delete(self, key):
+        self.cache.delete(key, noreply=False)
+
+
+class WrappedRedis(object):
```

Comment:
Should we do [more proper locking](https://redis.io/topics/distlock) in Redis? And if so, should we actually keep doing janky-locking in memcached?